qucs/qucs-filter/filter.cpp

```diff
@@ -47,195 +44,262 @@ double Filter::getE6value(double value)
 }
 
 // -----------------------------------------------------------------------
-double Filter::getNormValue(int No, tFilter *theFilter)
-{
-  if((No < 0) || (No >= theFilter->Order))
+double Filter::getNormValue(int No, tFilter *theFilter) {
+  if ((No < 0) || (No >= theFilter->Order))
     return 1.0;
 
-  switch(theFilter->Type) {
-    case TYPE_BESSEL:
-      return BesselValue(No, theFilter->Order);
-    case TYPE_BUTTERWORTH:
-      return ButterworthValue(No, theFilter->Order);
-    case TYPE_CHEBYSHEV:
-      if((theFilter->Order & 1) == 0) {
-        QMessageBox::critical(0, "Error", "Even order Chebyshev can't be realized with passive filters.");
-        return 2e30;
-      }
-      return ChebyshevValue(No, theFilter->Order, theFilter->Ripple);
+  switch (theFilter->Type) {
+  case TYPE_BESSEL:
+    return BesselValue(No, theFilter->Order);
+  case TYPE_BUTTERWORTH:
+    return ButterworthValue(No, theFilter->Order);
+  case TYPE_CHEBYSHEV:
+    if ((theFilter->Order & 1) == 0) {
+      QMessageBox::critical(
+          0, "Error",
+          "Even order Chebyshev can't be realized with passive filters.");
+      return 2e30;
+    }
+    return ChebyshevValue(No, theFilter->Order, theFilter->Ripple);
   }
 
   QMessageBox::critical(0, "Error", "Filter type not supported.");
   return 2e30;
 }
 
 // -----------------------------------------------------------------------
-double Filter::getQuadraticNormValues(int No, tFilter *theFilter, double &b)
-{
-  if((No < 0) || (No >= theFilter->Order))
+double Filter::getQuadraticNormValues(int No, tFilter *theFilter, double &b) {
+  if ((No < 0) || (No >= theFilter->Order))
     return 1.0;
 
-  switch(theFilter->Type) {
-    case TYPE_BESSEL:
-      return quadraticBesselValues(No, theFilter->Order, b);
-    case TYPE_BUTTERWORTH:
-      return quadraticButterworthValues(No, theFilter->Order, b);
-    case TYPE_CHEBYSHEV:
-      return quadraticChebyshevValues(No, theFilter->Order, theFilter->Ripple, b);
+  switch (theFilter->Type) {
+  case TYPE_BESSEL:
+    return quadraticBesselValues(No, theFilter->Order, b);
+  case TYPE_BUTTERWORTH:
+    return quadraticButterworthValues(No, theFilter->Order, b);
+  case TYPE_CHEBYSHEV:
+    return quadraticChebyshevValues(No, theFilter->Order, theFilter->Ripple, b);
   }
 
   QMessageBox::critical(0, "Error", "Filter type not supported.");
   return 2e30;
 }
 
 // -----------------------------------------------------------------------
-QString Filter::num2str(double Num)
-{
+QString Filter::num2str(double Num, units Unit) {
```

Comment:
I guess this is the function actually changed. But in the commit comment it's referenced with a different name :grin:

Reply:
Why did you choose to fix this by passing a *Unit* to `str2num`? Some filter types already have the correct units after the value (mostly the lumped elements ones, I think) while some other don't (the ones with transmission lines mostly, I think).
The type of element used is known in advance so there's no need to pass an additional parameter, just adding the correct  unit after the prefix, like already done for the lumped elements filters, should be enough.
And while you are at it, please re-add the space between the number and the prefix (see e.g. [here](https://www.bipm.org/utils/common/pdf/si_brochure_8_en.pdf), chapter 5.3.3. And, yes, it says also that *Ohm* should actually be written *ohm* but here we will continue with the Qucs tradition for the moment...)